### ISSUE TITLE
[BUG]: Return value of `create` different when using a transaction

### ISSUE THREAD

**Issue by donferi** [sensedeep/dynamodb-onetable#112]
**Describe the bug**
When passing a transaction to a `create`, all hidden attributes are returned and some types differ from when not using a transaction, eg: timestamps. I'm not sure if this is a bug or expected behavior. 

### Schema
```js
{
  name: 'MyTable',
  delimiter: '#',
  timestamps: true,
  isoDates: true,
  schema: {
    indexes: {
      primary: { hash: 'pk', sort: 'sk' },
    },
    models: {
      Team: {
        pk: { type: String, value: 'TEAM#${teamId}' },
        sk: { type: String, value: 'TEAM#${teamId}' },
        teamId: { type: String, required: true, uuid: true },
      }
    },
},
```

### Code
```js
const Team = table.getModel('Team');

const team = await Team.create({});

console.log(team);
};
```

The previous code works as expected, logs something like this:
```js
{
  teamId: 'N6GNGD8510JCJ5B298MPQ8BJDM',
  _type: 'Team',
  created: 2021-08-30T22:05:19.981Z,
  updated: 2021-08-30T22:05:19.981Z
}
```

<details>
  <summary>Logs</summary>
  
   ```
{
    "trace": {
        "cmd": {
            "ConditionExpression": "(attribute_not_exists(pk)) and (attribute_not_exists(sk))",
            "TableName": "Redacted",
            "Item": {
                "pk": {
                    "S": "TEAM#XI3THD8510K8SIIFC58AARX6JI"
                },
                "sk": {
                    "S": "TEAM#XI3THD8510K8SIIFC58AARX6JI"
                },
                "teamId": {
                    "S": "XI3THD8510K8SIIFC58AARX6JI"
                },
                "_type": {
                    "S": "Team"
                },
                "created": {
                    "S": "2021-08-30T22:27:51.834Z"
                },
                "updated": {
                    "S": "2021-08-30T22:27:51.834Z"
                }
            },
            "ReturnValues": "NONE"
        },
        "op": "put",
        "properties": {
            "pk": "TEAM#XI3THD8510K8SIIFC58AARX6JI",
            "sk": "TEAM#XI3THD8510K8SIIFC58AARX6JI",
            "teamId": "XI3THD8510K8SIIFC58AARX6JI",
            "_type": "Team",
            "created": "2021-08-30T22:27:51.834Z",
            "updated": "2021-08-30T22:27:51.834Z"
        },
        "params": {
            "parse": true,
            "high": true,
            "exists": false,
            "log": true,
            "checked": true
        }
    }
}

8/30/2021, 10:27:52 PM: info: OneTable result for "put" "Team"
{
    "trace": {
        "cmd": {
            "ConditionExpression": "(attribute_not_exists(pk)) and (attribute_not_exists(sk))",
            "TableName": "Redacted",
            "Item": {
                "pk": {
                    "S": "TEAM#XI3THD8510K8SIIFC58AARX6JI"
                },
                "sk": {
                    "S": "TEAM#XI3THD8510K8SIIFC58AARX6JI"
                },
                "teamId": {
                    "S": "XI3THD8510K8SIIFC58AARX6JI"
                },
                "_type": {
                    "S": "Team"
                },
                "created": {
                    "S": "2021-08-30T22:27:51.834Z"
                },
                "updated": {
                    "S": "2021-08-30T22:27:51.834Z"
                }
            },
            "ReturnValues": "NONE"
        },
        "op": "put",
        "properties": {
            "pk": "TEAM#XI3THD8510K8SIIFC58AARX6JI",
            "sk": "TEAM#XI3THD8510K8SIIFC58AARX6JI",
            "teamId": "XI3THD8510K8SIIFC58AARX6JI",
            "_type": "Team",
            "created": "2021-08-30T22:27:51.834Z",
            "updated": "2021-08-30T22:27:51.834Z"
        },
        "params": {
            "parse": true,
            "high": true,
            "exists": false,
            "log": true,
            "checked": true
        },
        "elapsed": 0.233,
        "items": [
            {
                "teamId": "XI3THD8510K8SIIFC58AARX6JI",
                "_type": "Team",
                "created": "2021-08-30T22:27:51.834Z",
                "updated": "2021-08-30T22:27:51.834Z"
            }
        ]
    }
}
   ```
</details>

However, this other code returns all attributes and notices how `created` and `updated` are strings instead of `Date`.
```js
const transaction = {};
const team = await Team.create({}, { transaction });

await table.transact('write', transaction);

console.log(team);
```

```js
{
  pk: 'TEAM#938RGD851099KEBBKHDT0TIC1T',
  sk: 'TEAM#938RGD851099KEBBKHDT0TIC1T',
  teamId: '938RGD851099KEBBKHDT0TIC1T',
  _type: 'Team',
  created: '2021-08-30T22:06:58.968Z',
  updated: '2021-08-30T22:06:58.968Z'
}
```

I can't get any logs out of this one, not sure why.

**Expected behavior**
I would expect the result of the second snippet to match the result of the first one.

**Environment (please complete the following information):**
 - OS: Mac OS
 - Node Version: 14.x.x

**Comment by mobsense** [comment]
That is a bug, we need to do the following in Model.run()

                return this.transformReadItem(op, properties, properties, params)

We'll fix that.

Note: when you do the Team.create() is is NOT actually creating the Team at that point, and the DynamoDB transaction itself does not return properties.

**Comment by mobsense** [comment]
Fix planned for 1.7.5 today
